simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,400 @@
+
+    parser.add_argument(
+        '--randomSeed',
+        default = np.random.randint(DEFAULT_MAX_RAND_INT),
+        type = int,
+        help = 'Seed for the random number generator used in the script.')
+
+    parser.add_argument(
+        '--targetSplit',
+        default = DEFAULT_TARGET_SPLIT,
+        type = checkValidSplit,
+        help = 'Fraction of the truth data to be used as targets.')
+
+    args = parser.parse_args()
+    return args
+
+
+def setRandomSeed(seed):
+    """
+    Set the seed for all random number generators in the script.
+    """
+    np.random.seed(seed)
+
+
+def getTruncatedNormal(mean, sd, low, upp):
+    """
+    Return truncated normal distribution.
+    """
+    return truncnorm(
+        (low - mean) / sd, (upp - mean) / sd, loc = mean, scale = sd)
+
+
+class Person():
+    """
+    Person class for the dataset.
+    """
+
+    def __init__(self):
+        self.index = None
+        self.globalLikes = None
+        self.lived = None
+        self.localLikes = None
+
+
+class DataGen():
+    """
+    Data generator class for simple-acquaintances example.
+    """
+
+    def __init__(
+            self,
+            numberOfPeople,
+            numberOfPlaces,
+            numberOfGlobal,
+            numberOfLocal,
+            placesLivedMean,
+            placesLivedSD,
+            localLikesVariance,
+            targetSplit,
+            randomSeed,
+            outputDir):
+        self.numberOfPeople = numberOfPeople
+        self.places = [i for i in range(numberOfPlaces)]
+        self.globalThings = [i for i in range(numberOfGlobal)]
+        self.localThings = [i for i in range(numberOfLocal)]
+        self.placesLivedMean = placesLivedMean
+        self.placesLivedSD = placesLivedSD
+        self.placesLivedUpper = numberOfPlaces
+        self.localLikesVariance = localLikesVariance
+        self.targetSplit = targetSplit
+        self.randomSeed = randomSeed
+        self.outputDir = outputDir
+
+        self.placesAffectingLocalThings = {}
+        self.__generatePALT()  # Fill in placesAffectingLocalThings.
+
+        setRandomSeed(self.randomSeed)
+
+    def __generatePALT(self):
+        """
+        Generate porbabilities of places affecting the likeablity of local things.
+
+        eg: [Place1:[LThing1, Lthing2], Place2:[LThing1,LThing2]]
+        """
+        for i in self.places:
+            self.placesAffectingLocalThings[i] = np.random.uniform(
+                0, 1, len(self.localThings))
+
+    def __getLikeability(self, person):
+        """
+        Return likeability of localThings based on the places that a person has lived in.
+        """
+        likeability = {}
+        for thing in self.localThings:
+            likeability[thing] = 0
+            for place in person.lived:
+                truncnorm_gen = getTruncatedNormal(self.placesAffectingLocalThings[place][thing],
+                                                    self.localLikesVariance, 0, 1)
```

Comment:
What's the deal with the spacing here?
It is not a constant indent (my favored method), nor aligned (my unflavored, but tolerated method).

Reply:
Sorry about that I totally forgot to change it. I was trying to use an extension in VSCode to indent my code and messing around with its settings.